### PR TITLE
err_set_debug(): Prevent possible recursion on malloc failure

### LINKED ISSUE #19331: Possible endless recursion in malloc error reporting
Due to the changes from 5639ee79bdc905c1b7ed308fcd29e515e4f01a50
there is now a possible endless recursion if malloc suddenly starts to fail here (in CRYPTO_malloc):
https://github.com/openssl/openssl/blob/fcae2ae4f675def607d338b7945b9af1dd9bb746/crypto/mem.c#L206
-> and then (in ERR_set_debug):
https://github.com/openssl/openssl/blob/fcae2ae4f675def607d338b7945b9af1dd9bb746/crypto/err/err_blocks.c#L37
-> and then (in err_set_debug):
https://github.com/openssl/openssl/blob/fcae2ae4f675def607d338b7945b9af1dd9bb746/crypto/err/err_local.h#L60-L66
-> which is a macro:
https://github.com/openssl/openssl/blob/fcae2ae4f675def607d338b7945b9af1dd9bb746/include/openssl/crypto.h.in#L110-L111
-> and then (in CRYPTO_strdup):
https://github.com/openssl/openssl/blob/fcae2ae4f675def607d338b7945b9af1dd9bb746/crypto/o_str.c#L27
-> so this will enter a loop now

### PR DESCRIPTION
**Pull request by t8m**
Fixes #19331

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: crypto/err/err_local.h [hunk 1]
```diff
@@ -56,8 +56,11 @@ static ossl_inline void err_set_debug(ERR_STATE *es, size_t i,
...
-    else
-        es->err_file[i] = OPENSSL_strdup(file);
+    else if ((es->err_file[i] = CRYPTO_malloc(strlen(file) + 1,
+                                              NULL, 0)) != NULL)
+        /* We cannot use OPENSSL_strdup due to possible recursion */
+        strcpy(es->err_file[i], file);
+
     es->err_line[i] = line;
     OPENSSL_free(es->err_func[i]);
     if (fn == NULL || fn[0] == '\0')
```
Comment (bernd-edlinger) [inline crypto/err/err_local.h:66]: Haha...
and what if the malloc fails in the OPENSSL_strdup below?

Comment (t8m) [reply crypto/err/err_local.h:66]: The `fn == NULL` when ERR_set_debug() is invoked from CRYPTO_malloc()

**Comment by t8m** [comment]
@paulidale I had to add the `#include`. Still OK?

**Review by paulidale — APPROVED** [review]
still fine

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for the reviews.

**Pull request closed without merging**